main.nf

```diff
@@ -1149,6 +1153,17 @@ workflow.onComplete {
     if(workflow.container) email_fields['summary']['Docker image'] = workflow.container
     email_fields['skipped_poor_alignment'] = skipped_poor_alignment
 
+    // On success try attach the multiqc report
+    def mqc_report = null
+    def find_mqc_report = {it.eachFileMatch(~/.*multiqc_report.html/) {mqc_report = it}}
+    try {
+        if (workflow.success) {
+            find_mqc_report(new File("${params.outdir}/MultiQC/"))
```

Comment:
Ok interesting! I assumed that you would use the `multiqc_report` output channel, and was curious as to how that would work.

Note this solution isn't my _favourite_  option. If you run the pipeline multiple times with the same output directory, the results will be added to the previous run (overwriting if the same filename is found). As MultiQC filenames are often dynamically generated, there could be an edge case here where multiple / incorrect MultiQC reports are attached to the email. Edge case only though.

Did you look into the channel thing at all? It may be possible to pull the raw filename from that directly, don't know.

Reply:
Yes, I am not fully satisfied either. As you mentioned, it will just take one file (which would be randon) so there is a high chance that old file gets attached.

That is why I started by using `multiqc_output` channel. But I just gave up while trying to get what I want from a "channel".
(But while writing this comment I had a lightbulb 💡, so let me try that 😃)